torchgeo/trainers/regression.py

```diff
@@ -26,36 +26,74 @@
 
 
 class RegressionTask(pl.LightningModule):
-    """LightningModule for training models on regression datasets."""
+    """LightningModule for training models on regression datasets.
+
+    Supports any available `Timm model
+    <https://rwightman.github.io/pytorch-image-models/>`_
+    as an architecture choice. To see a list of available pretrained
+    models, you can do:
+
+    .. code-block:: python
+
+        import timm
+        print(timm.list_models(pretrained=True))
+    """
 
     def config_task(self) -> None:
         """Configures the task based on kwargs parameters."""
+        in_channels = self.hyperparams["in_channels"]
         model = self.hyperparams["model"]
-        pretrained = self.hyperparams["pretrained"]
-
-        if parse(torchvision.__version__) >= parse("0.13"):
-            if pretrained:
-                kwargs = {
-                    "weights": getattr(
-                        torchvision.models, f"ResNet{model[6:]}_Weights"
-                    ).DEFAULT
-                }
+
+        imagenet_pretrained = False
+        custom_pretrained = False
+        if self.hyperparams["weights"] and not os.path.exists(
+            self.hyperparams["weights"]
+        ):
+            if self.hyperparams["weights"] not in ["imagenet", "random"]:
+                raise ValueError(
+                    f"Weight type '{self.hyperparams['weights']}' is not valid."
+                )
             else:
-                kwargs = {"weights": None}
+                imagenet_pretrained = self.hyperparams["weights"] == "imagenet"
+            custom_pretrained = False
+        else:
+            custom_pretrained = True
+
+        # Create the model
+        valid_models = timm.list_models(pretrained=True)
```

Comment:
Shouldn't this be:
```suggestion
        valid_models = timm.list_models(pretrained=False)
```
That way you can still use a model without pre-trained weights, or with custom pre-trained weights.